### ISSUE TITLE
Add Foliate as a .deb package from GitHub instead of PPA

### ISSUE THREAD

**Issue by sbstn87** [wimpysworld/deb-get#685]
- [X] I have checked that this app does not already have a pending request and has not previously been declined.
 - App Name: Foliate
 - Download URL: https://github.com/johnfactotum/foliate/releases
 - Website URL: https://johnfactotum.github.io/foliate/
 - App Summary:  A simple and modern GTK eBook reader.

App is already available from deb-get but is sourced from PPA, which makes it impossible to install it on non-Ubuntu distros. Switching to a .deb package would make it available for all Debian based distros, not only *buntus.

**Comment by silentJET85** [comment]
I was going to make a PR to change this to the Github release, but it doesn't actually install on Bookworm due to depending on a newer version of gir1.2-gtk-4.0.
It would probably install on Trixie/Sid, but those already have a native package for Foliate that is nearly up-to-date, so that would be redundant.

**Comment by sbstn87** [comment]
When Trixie is released, Foliate will be stuck again at the current version for the next two years, so I suppose adding it to deb-get still will be useful.

**Comment by philclifford** [comment]
I'm not a fan of having a GitHub source and a PPA source treated as a web source. But we can (have) chose appropriate source based on the host OS where that benefits users. I wouldn't want to de-support jammy users, and particularly not just to introduce support for Trixie/Sid where Bookworm is not supported. I would like to add support via GitHub releases for Debians and derivatives that can be so supported.

**Comment by silentJET85** [comment]
It seems like you're saying it's okay to have different installation methods in a single package, but I was previously told not to do that. Is the prettylist thing mentioned in #521 not an issue anymore? Just trying to wrap my head around seemingly conflicting information. <br><br>

 
              A package should not have more than one installation method, since it would produce different outputs to `prettylist` in different systems. I think that in this case the best solution would be to replace the PPA with the GitHub releases method in all cases.

_Originally posted by @natanjunges in https://github.com/wimpysworld/deb-get/issues/521#issuecomment-1246096506_


              Please consolidate this change so Ubuntu and Debian install via the same mechanism.

_Originally posted by @flexiondotorg in https://github.com/wimpysworld/deb-get/pull/521#pullrequestreview-1111928110

**Comment by philclifford** [comment]
@silentJET85 you are correct - my opinion now, given the current circumstances and the discussion around this package, differs from that held by natan and @flexiondotorg regarding #521.  It is true that that output of `prettylist` will be different on Ubuntu-ish and Debian-ish systems.  Personally I have no problem with that (and see the balance between what I see as somewhat pointless uniformity and potential benefits to both Ubuntu and Debian users tipped very much in favour of the latter).  Martin may have sound reasons why my opinion is problematic, or he may agree that different appropriate sources may be the best approach for some upstreams.  If he sees it as problematic he may wish to review #1293 which I merged last week, since it essentially also supports PPA-sourced (disguised as APT ) for Ubuntu and APT sourced from OBS for Debians.  Obviously that shows the same source in `prettylist` on all systems where supported but in principle there are different sources for different hosts.  Sadly since in this instance the Debian hosts will be using GitHub we'll always have a different source type.

I'll try and create another PR myself for the split source approach and arrange for either merge to close the other so @flexiondotorg can comment or make the choice between

1. Only support Ubuntus via PPA (i.e. do nothing and close the issue and PRs)
2. Add Debian support and remove support for Jammy and Focal (this PR as it stands)
3. Add Debian support via GitHub but leave Ubuntus as PPA-sourced (depending on a new PR or tweaking this one)

**Comment by silentJET85** [comment]
I agree with you. Option 3 seems best.

In doing some more testing, I've found that this won't always install on Sid because of #674. 
Depending on how it was installed, Sid/Trixie's `UPSTREAM_CODENAME` is being reported as "bookworm" or "13."
I'm pretty sure I know how to fix it. I'll make a PR when I get a chance.
